photon-lib/src/main/scala/com/linkedin/photon/ml/util/VectorUtils.scala

```diff
@@ -284,4 +286,15 @@ object VectorUtils {
 
       set
   }
+
+  /**
+   * Element-wise inversion of a [[Vector]], if an element is smaller than MathConst.EPSILON, the inversion will be
+   * replaced by zeroReplacedVal.
+   *
+   * @param vector The [[Vector]] to invert
+   * @param zeroReplacedVal The replaced value when inverting an element close to 0
+   * @return The inverted [[Vector]]
+   */
+  def invertVectorWithZeroHandler(vector: Vector[Double], zeroReplacedVal: Double): Vector[Double] =
+    vector.map(v => if (v > MathConst.EPSILON) 1.0 / v else zeroReplacedVal)
```

Comment:
Shouldn't we checkout the absolute value of `v`?

Reply:
Although variance is always non-negative, but I suggest to do it general for this util function.

Reply:
fixed